yarn-project/end-to-end/src/fixtures/cheat_codes.ts

```diff
@@ -127,8 +127,69 @@ export class L1CheatCodes {
     this.logger(`Loaded state from ${fileName}`);
   }
 
-  // Good basis for the remaining functions:
-  // https://github.com/foundry-rs/foundry/blob/master/anvil/core/src/eth/mod.rs
+  /**
+   * Load the value at a storage slot of a contract address on L1
+   * @param contract - The contract address
+   * @param slot - The storage slot
+   * @returns - The value at the storage slot in hex
+   */
+  public async getStorageAt(contract: EthAddress, slot: `0x${string}`): Promise<`0x${string}`> {
+    const res = await this.rpcCall('eth_getStorageAt', [contract.toString(), slot, 'latest']);
+    return res.result;
+  }
+
+  /**
+   * Set the value at a storage slot of a contract address on L1
+   * @param contract - The contract address
+   * @param slot - The storage slot
+   * @param value - The value to set the storage slot to
+   */
+  public async setStorageAt(contract: EthAddress, slot: `0x${string}`, value: `0x${string}`): Promise<void> {
+    const res = await this.rpcCall('anvil_setStorageAt', [contract.toString(), slot, value]);
+    if (res.error) throw new Error(`Error setting storage at ${slot}: ${res.error.message}`);
```

Comment:
Include the contract in the error as well, just to make it clear what contract it was that has issues.